packages/form-layout/test/form-item.test.js

```diff
@@ -70,17 +100,83 @@ describe('form-item', () => {
     });
   });
 
-  describe('label click', () => {
-    it('should focus input', () => {
-      const spy = sinon.spy(input, 'focus');
-      label.click();
-      expect(spy.called).to.be.true;
+  describe('aria-labelledby', () => {
+    describe('input', () => {
+      beforeEach(async () => {
+        item = fixtureSync(`
+          <vaadin-form-item>
+            <label slot="label">Label</label>
+            <input aria-labelledby="custom-id">
+          </vaadin-form-item>
+        `);
+        label = item.querySelector('label');
+        input = item.querySelector('input');
+        await nextFrame();
+      });
+
+      it('should link the label to the input', () => {
+        expect(input.getAttribute('aria-labelledby')).to.equal(`custom-id ${label.id}`);
+      });
+
+      it('should link the label only to the first input', async () => {
+        const newInput = document.createElement('input');
+        item.appendChild(newInput);
+        await nextFrame();
+        expect(input.getAttribute('aria-labelledby')).to.equal(`custom-id ${label.id}`);
+        expect(newInput.hasAttribute('aria-labelledby')).to.be.false;
+      });
+
+      it('should link the label to the new input when replacing the input', async () => {
```

Comment:
As we have tests for `appendChild` and `replaceChild`, let's consider testing `inertBefore` also.

Reply:
Added.